src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php

```diff
@@ -41,6 +43,8 @@ public function __construct(KernelInterface $kernel)
      */
     public function parse($controller)
     {
+        @trigger_error(sprintf('The %s class is deprecated since version 4.1 and will be removed in 5.0.', __CLASS__), E_USER_DEPRECATED);
```

Comment:
why not put the trigger on top of the file?

Reply:
because we still use the class and service internally. but the trigger only matters when it gets executed